packages/evo/src/wrap/js-engine/Date.ts

```diff
@@ -0,0 +1,203 @@
+    } else {
+      this._timestamp = args[0].timestamp;
+    }
+  }
+
+  static now() {
+    return __wrap_subinvoke("plugin/datetime", "now", clean(undefined)).value;
+  }
+
+  static parse(date) {
+    return __wrap_subinvoke("plugin/datetime", "parse", clean({ date })).value;
+  }
+
+  static UTC(year, month, day, hour, minute, second, millisecond) {
+    const args = clean({ year, month, day, hour, minute, second, millisecond });
+    return __wrap_subinvoke("plugin/datetime", "UTC", args).value;
+  }
+
+  getDate() {
+    return __wrap_subinvoke("plugin/datetime", "getDate", clean({ timestamp: this._timestamp })).value;
+  }
+
+  getDay() {
+    return __wrap_subinvoke("plugin/datetime", "getDate", clean({ timestamp: this._timestamp })).value;
+  }
+
+  getFullYear() {
+    return __wrap_subinvoke("plugin/datetime", "getFullYear", clean({ timestamp: this._timestamp })).value;
+  }
+
+  getHours() {
+    return __wrap_subinvoke("plugin/datetime", "getHours", clean({ timestamp: this._timestamp })).value;
+  }
+
+  getMilliseconds() {
+    return __wrap_subinvoke("plugin/datetime", "getMilliseconds", clean({ timestamp: this._timestamp })).value;
+  }
+
+  getMinutes() {
+    return __wrap_subinvoke("plugin/datetime", "getMinutes", clean({ timestamp: this._timestamp })).value;
+  }
+
+  getMonth() {
+    return __wrap_subinvoke("plugin/datetime", "getMonth", clean({ timestamp: this._timestamp })).value;
+  }
+
+  getSeconds() {
+    return __wrap_subinvoke("plugin/datetime", "getSeconds", clean({ timestamp: this._timestamp })).value;
+  }
+
+  getTime() {
+    return __wrap_subinvoke("plugin/datetime", "getTime", clean({ timestamp: this._timestamp })).value;
+  }
+
+  getTimezoneOffset() {
+    return __wrap_subinvoke("plugin/datetime", "getTimezoneOffset", clean({ timestamp: this._timestamp })).value;
+  }
+
+  getUTCDate() {
+    return __wrap_subinvoke("plugin/datetime", "getUTCDate", clean({ timestamp: this._timestamp })).value;
+  }
+
+  getUTCDay() {
+    return __wrap_subinvoke("plugin/datetime", "getUTCDay", clean({ timestamp: this._timestamp })).value;
+  }
+
+  getUTCFullYear() {
+    return __wrap_subinvoke("plugin/datetime", "getUTCFullYear", clean({ timestamp: this._timestamp })).value;
+  }
+
+  getUTCHours() {
+    return __wrap_subinvoke("plugin/datetime", "getUTCHours", clean({ timestamp: this._timestamp })).value;
+  }
+
+  getUTCMilliseconds() {
+    return __wrap_subinvoke("plugin/datetime", "getUTCMilliseconds", clean({ timestamp: this._timestamp })).value;
+  }
+
+  getUTCMinutes() {
+    return __wrap_subinvoke("plugin/datetime", "getUTCMinutes", clean({ timestamp: this._timestamp })).value;
+  }
+
+  getUTCMonth() {
+    return __wrap_subinvoke("plugin/datetime", "getUTCMonth", clean({ timestamp: this._timestamp })).value;
+  }
+
+  getUTCSeconds() {
+    return __wrap_subinvoke("plugin/datetime", "getUTCSeconds", clean({ timestamp: this._timestamp })).value;
+  }
+
+  setDate(day) {
```

Comment:
These set methods all modify the date object, which is not the case for our shims

Reply:
Could you clarify what change you're requesting? The issue is that the Date class is a class, or that it is stateful? It would help if you could tell me more about the limitations of shims. Do they have to be static methods?

Reply:
This date class should be stateful, but it's not 
If you do the following in standard JS:
```javascript
let time = new Date();
time.setSeconds(30);
console.log(time.getSeconds());
```
It will print 30 (cause we set it to 30)

If you do the same with these shims, it will not (cause your set methods don't use state)

Reply:
Okay, I see what you're saying. Thanks for the clarification. I pushed a fix.

Reply:
It looks like that'll solve the issue, but now the methods no longer return the timestamp (as they should per the API reference)

Reply:
I'm just swapping one error for another haha. I pushed a fix.